x-pack/plugins/code/public/components/main/commit_history.tsx

```diff
@@ -75,11 +81,24 @@ export const CommitHistoryLoading = () => (
   </div>
 );
 
-export const CommitHistory = (props: {
+export const PageButtons = (props: { disabled: boolean; onClick: () => void }) => (
+  <EuiFlexGroup justifyContent="spaceAround">
+    <EuiFlexItem grow={false}>
+      <EuiButton onClick={props.onClick} iconType="arrowDown" isDisabled={props.disabled} size="s">
```

Comment:
Instead of swapping the button out for a loading spinner and then back to a button. You could use the `isLoading` prop on the button that will disable it and replace the icon with the spinner. Just a thought.